app/views/spree/admin/orders/_invoice_table.html.haml

```diff
@@ -10,19 +10,35 @@
       %th{:align => "right", :width => "15%"}
         %h4= t(:invoice_column_price)
   %tbody
-    - @order.line_items.sort_by{ |li| li.product.name }.each do |item|
-      %tr
-        %td
-          = render 'spree/shared/line_item_name', line_item: item
-          %br
-          %small
-            %em= raw(item.variant.product.supplier.name)
-        %td{:align => "right"}
-          = item.quantity
-        %td{:align => "right"}
-          = item.included_tax > 0 ? item.display_included_tax : ""
-        %td{:align => "right"}
-          = item.display_amount_with_adjustments
+    - if @order.distributor.preferred_invoice_order_by_supplier == true
+      - @order.line_items.sort_by{ |li| li.supplier.name }.each do |item|
```

Comment:
I think that it would still be useful to order by product name, too:

1. SupplierA - Apples
2. SupplierA - Bananas
3. SupplierB - Apples
4. SupplierB - Bananas

Otherwise you get random product orders within each supplier.